gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java

```diff
@@ -381,6 +383,15 @@ protected String defaultProject() {
     return projectId != null ? projectId : googleCloudProjectId();
   }
 
+  private static String activeGoogleCloudConfig(File configDir) {
+    try {
+      return Files.readFirstLine(new File(configDir, "active_config"), Charset.defaultCharset());
```

Comment:
When you have only `default` config this file contains `default`. This always points to the active config.